vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -36,11 +36,15 @@
 }
 
 
-def _filter_between(series: pd.Series, value: List[float]) -> pd.Series:
+def _filter_between(series: pd.Series, value: Union[List[float], List[str]]) -> pd.Series:
+    if is_datetime64_any_dtype(series):
+        value = pd.to_datetime(value)
```

Comment:
Are we catching the case where this conversion does not work?

Reply:
Or a different thought: when does this even receive a string? The [datepicker](https://github.com/mckinsey/vizro/blob/926037ce8048a79a50299e68c4b5f95e3a7d2b70/vizro-core/src/vizro/models/_components/form/date_picker.py#L41) only allows `date`, so why would this even be a string?

Reply:
If `is_datetime64_any_dtype(series)` is True, this means that the `value` comes from the DatePicker UI component, which is always in string format that can be converted to a datetime object. We know the value comes from the DatePicker UI component because `is_datetime64_any_dtype(series)` is True, and we only allow the DatePicker selector to be set.

Reply:
> The [datepicker](https://github.com/mckinsey/vizro/blob/926037ce8048a79a50299e68c4b5f95e3a7d2b70/vizro-core/src/vizro/models/_components/form/date_picker.py#L41) only allows date, so why would this even be a string?

Yes, the DatePicker `value` allows values that are convertible to `date` see the [link](https://docs.pydantic.dev/latest/api/standard_library_types/#datetime-types), but inside the callback context, selected value always comes in the string format. More exactly in the 'yyyy-mm-dd' (or ['yyyy-mm-dd', 'yyyy-mm-dd']) format.